src/main/java/org/jabref/preferences/AiPreferences.java

```diff
@@ -276,4 +436,115 @@ public void onEmbeddingsParametersChange(Runnable runnable) {
             }
         });
     }
+
+    public void listenToChatModels(Runnable runnable) {
+        openAiChatModel.addListener((observableValue, oldValue, newValue) -> {
+            if (!newValue.equals(oldValue)) {
+                runnable.run();
+            }
+        });
+
+        mistralAiChatModel.addListener((observableValue, oldValue, newValue) -> {
+            if (!newValue.equals(oldValue)) {
+                runnable.run();
+            }
+        });
+
+        huggingFaceChatModel.addListener((observableValue, oldValue, newValue) -> {
+            if (!newValue.equals(oldValue)) {
+                runnable.run();
+            }
+        });
+    }
+
+    public void listenToApiTokens(Runnable runnable) {
+        openAiApiToken.addListener((observableValue, oldValue, newValue) -> {
+            if (!newValue.equals(oldValue)) {
+                runnable.run();
+            }
+        });
+
+        mistralAiApiToken.addListener((observableValue, oldValue, newValue) -> {
+            if (!newValue.equals(oldValue)) {
+                runnable.run();
+            }
+        });
+
+        huggingFaceApiToken.addListener((observableValue, oldValue, newValue) -> {
+            if (!newValue.equals(oldValue)) {
+                runnable.run();
+            }
+        });
+    }
+
+    public void listenToApiBaseUrls(Runnable runnable) {
+        openAiApiBaseUrl.addListener((observableValue, oldValue, newValue) -> {
+            if (!newValue.equals(oldValue)) {
+                runnable.run();
+            }
+        });
+
+        mistralAiApiBaseUrl.addListener((observableValue, oldValue, newValue) -> {
+            if (!newValue.equals(oldValue)) {
+                runnable.run();
+            }
+        });
+
+        huggingFaceApiBaseUrl.addListener((observableValue, oldValue, newValue) -> {
+            if (!newValue.equals(oldValue)) {
+                runnable.run();
+            }
+        });
+    }
+
+    public String getSelectedChatModel() {
+        return switch (aiProvider.get()) {
+            case OPEN_AI ->
+                    openAiChatModel.get();
+            case MISTRAL_AI ->
+                    mistralAiChatModel.get();
+            case HUGGING_FACE ->
+                    huggingFaceChatModel.get();
+        };
+    }
+
+    public String getSelectedApiToken() {
+        if (!enableAi.get()) {
+            return "";
+        }
+
+        retrieveTokens();
```

Comment:
Now, at each call, the key store is asked.

It should be asked at the first call, then the stored value should be used.

Reason:

1. We assume that a key change is done within JabRef and not externally
2. Each call to the key store could lead to an ask of the user to give access to the key store
3. The key is stored in memory 

Alternatively, always call referencesService.getApiKeyForAiProvider and do not store it at all.